src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -692,6 +688,59 @@ void PolicyHandler::LinkAppsToDevice() {
   }
 }
 
+bool PolicyHandler::IsAppSuitableForPolicyUpdate(
+    const Applications::value_type value) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const uint32_t app_id = value->app_id();
+
+  if (!value->IsRegistered()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Application " << app_id << " is not marked as registered.");
+    return false;
+  }
+
+  LOG4CXX_DEBUG(logger_,
+                "Application " << app_id << " marked as registered."
+                                            "Checking its parameters.");
+
+  DeviceParams device_params = GetDeviceParams(
+      value->device(),
+      application_manager_.connection_handler().get_session_observer());
+
+  const bool is_device_allowed = (kDeviceAllowed ==
+                                  policy_manager_->GetUserConsentForDevice(
+                                      device_params.device_mac_address));
+
+  LOG4CXX_DEBUG(logger_,
+                "Is device " << device_params.device_mac_address << " allowed "
+                             << std::boolalpha << is_device_allowed);
+
+  if (!is_device_allowed) {
+    return false;
+  }
+
+  return true;
```

Comment:
@LevchenkoS , can replace this code 
`if (!is_device_allowed) {
    return false;
  }
  return true;`
with `return is_device_allowed`